src/dag_union.rs

```diff
@@ -0,0 +1,109 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use crate::digraph::PyDiGraph;
+use petgraph::algo;
+use petgraph::graph::EdgeIndex;
+use pyo3::prelude::*;
+use pyo3::Python;
+use std::cmp::Ordering;
+use std::collections::BTreeMap;
+use std::collections::HashSet;
```

Comment:
We use hashbrown hashset type else where in retworkx. It's basically the same as the version in std::collections except that it's faster because it uses a different hashing algorithm by default and that it supports parallelization with rayon (which probably doesn't matter here)

```suggestion
use hashbrown::HashSet;
```

Reply:
fixed